src/stac_utils/validators.py

```diff
@@ -0,0 +1,75 @@
+import re
```

Comment:
given the exact content of this, i feel like maybe it should be normalize.py instead?

should also get test coverage, docstrings, and added to docs/index.rst